cross_region_replication_latency/README.md

```diff
@@ -0,0 +1,83 @@
+General Info and Prerequisites
+==============================
+
+This scripts enables you to measure Scylla's cross DC/region replication latency on Cloud or on-prem deployments.
+
+In this specific code example we are using a single writer in us-east (N. Virginia) to write a single partition using CL=LOCAL_ONE, which we then read from us-west (Oregon) using a single reader.
+
+The write rate is ~5 partitions per second. We calculate the replication latency time on the reader’s end by subtracting the partition's insert time from the partition's now() time at the moment it is read.
```

Comment:
Not sure I get it
How would the reader know when the data was updated?
Does it loop on the cell, doing a select every time?
If so, the loop add a lot of noise

Reply:
all scylla nodes clock is synced , I used Chrony for that (mentioned in the doc prereq).
We have the partition insert time in the DB and we have the partition now() time in the DB at the moment it is read from the reader's end hence we subtract them to get the diff which is the replication latency.

The flow works like this:
1. Writer (DC-1) inserts partition to the test table
2. Reader (DC-2 reads the partition with the now() time from the DB
Example:
```
cqlsh> SELECT id, insertion_date, now() AS now, some_data, WRITETIME (some_data) FROM replicated.test;

 id                                   | insertion_date                       | now                                  | some_data | writetime(some_data)
--------------------------------------+--------------------------------------+--------------------------------------+-----------+----------------------
 993c802e-6978-11e8-a1eb-0a77d2c75dc0 | 993835f0-6978-11e8-a2d4-000000000001 | 690098d0-697f-11e8-a747-000000000000 |     dummy |     1528282732267178
```

3. Reader updated count table (counter)
4. Writer reads counter table and truncates both tables
5. Steps 1-4 repeat

Reply:
Are you using client or server timestamp?

Reply:
Server timestamp for both insert (writer) and now() time when we read it (from the reader's end).

Reply:
NOW() return the time of the read, not the time the data updated the in remote DC.
How do you know the update was not earlier?

Reply:
That's the best I could think of, and how Dor wanted to measure it. from
the Writer at one end (DC-1) to the reader from the second end (DC-2).